server/src/main/java/org/elasticsearch/snapshots/RepositoryIntegrityHealthIndicatorService.java

```diff
@@ -73,78 +104,160 @@ public String name() {
+        private final Set<String> nodesWithUnknownRepos = new HashSet<>();
+        private final Set<String> invalidRepositories = new HashSet<>();
+        private final Set<String> nodesWithInvalidRepos = new HashSet<>();
+        private final HealthStatus healthStatus;
+
+        private RepositoryHealthAnalyzer(
+            ClusterState clusterState,
+            List<RepositoryMetadata> repositories,
+            Map<String, RepositoriesHealthInfo> repositoriesHealthByNode
+        ) {
+            this.clusterState = clusterState;
+            this.totalRepositories = repositories.size();
+            this.corruptedRepositories = repositories.stream()
+                .filter(repository -> repository.generation() == RepositoryData.CORRUPTED_REPO_GEN)
+                .map(RepositoryMetadata::name)
+                .toList();
+
+            repositoriesHealthByNode.forEach((nodeId, healthInfo) -> {
+                unknownRepositories.addAll(healthInfo.unknownRepositories());
+                if (healthInfo.unknownRepositories().isEmpty() == false) {
+                    nodesWithUnknownRepos.add(nodeId);
+                }
+                invalidRepositories.addAll(healthInfo.invalidRepositories());
+                if (healthInfo.invalidRepositories().isEmpty() == false) {
+                    nodesWithInvalidRepos.add(nodeId);
+                }
+            });
+
+            if (corruptedRepositories.isEmpty() == false
+                || unknownRepositories.isEmpty() == false
+                || invalidRepositories.isEmpty() == false) {
+                healthStatus = YELLOW;
+            } else if (repositoriesHealthByNode.isEmpty()) {
+                healthStatus = UNKNOWN;
+            } else {
+                healthStatus = GREEN;
+            }
+        }
+
+        public HealthStatus getHealthStatus() {
+            return healthStatus;
+        }
+
+        public String getSymptom() {
+            if (healthStatus == GREEN) {
+                return ALL_REPOS_HEALTHY;
+            } else if (healthStatus == UNKNOWN) {
+                return NO_REPO_HEALTH_INFO;
+            }
+
+            return "Detected "
+                + Stream.of(
+                    generateSymptomString("corrupted", corruptedRepositories.size()),
+                    generateSymptomString("unknown", unknownRepositories.size()),
+                    generateSymptomString("invalid", invalidRepositories.size())
+                ).filter(Objects::nonNull).collect(Collectors.joining(", and "))
+                + ".";
+        }
+
+        private static String generateSymptomString(String type, long size) {
+            if (size == 0) {
+                return null;
+            }
+
+            return String.format(Locale.ROOT, "[%d] %s snapshot repositor%s", size, type, size > 1 ? "ies" : "y");
+        }
+
+        public HealthIndicatorDetails getDetails(boolean verbose) {
+            if (verbose == false || healthStatus == UNKNOWN) {
```

Comment:
Why are the details empty when the `healthStatus` is `unknown`? There are still details we can provide, like:

- `total_repositories`
- `corrupted_repositories` etc

Reply:
Good point. Updated and added a test for `UNKNOWN` status`.